components/projects/NewProject/NewCustomer.tsx

```diff
@@ -9,28 +11,92 @@ import {
     ModalContent,
     ModalBody,
     ModalFooter,
+    Box,
 } from '@chakra-ui/react';
-import React from 'react';
+import React, { useState } from 'react';
+import { components } from '@/lib/types/api';
+import * as fetch from '@/lib/utils/fetch';
+import { FormStatus } from './reducer';
+import { useSWRConfig } from 'swr';
+import ErrorAlert from '@/components/common/ErrorAlert';
+
+type StateType = {
+    name: string;
+    description: string;
+    formStatus: FormStatus;
+};
 
 function NewCustomer(): JSX.Element {
     const { isOpen, onOpen, onClose } = useDisclosure();
+    const [state, setState] = useState<StateType>({ name: '', description: '', formStatus: FormStatus.IDLE });
+    const url = `${process.env.NEXT_PUBLIC_API_URL}/customer`;
+    const { mutate } = useSWRConfig();
+
+    const handleSubmit = async (e: React.MouseEvent) => {
+        e.preventDefault();
+
+        const createCustomerRequest: components['schemas']['CustomerDTO'] = {
+            name: state.name,
+            description: state.description,
+        };
+        setState({ name: state.name, description: state.description, formStatus: FormStatus.LOADING });
```

Comment:
What's the purpose of the first two `name: state.name` and `description: state.description`? I don't think setting the state to the same value does really anything and can be omitted

Reply:
I think the state will only contain the formStatus value unless name and description are explicitly given

Reply:
Strange behavior. `setState` should only affect the specified fields, maybe @robertsiipola know if there's something wrong with the state definition. 

Anyhoo, maybe you can use the spread operator with the existing state combined with setting the fields that are actually being change, e.g.
```typescript
setState({
    ...state,
    formStatus.LOADING
})
```
to only change the `formStatus` value?

Reply:
> setState should only affect the specified fields

I actually know jack shit about the hooks way of managing state. Apparently some even use own hooks for each state parameter. But spreading the existing state should be a good way to reduce some repetitive code

Reply:
spread operator put to use in 296a601